synapse/storage/data_stores/main/cache.py

```diff
@@ -66,7 +71,24 @@ def get_all_updated_caches_txn(txn):
         )
 
     def process_replication_rows(self, stream_name, instance_name, token, rows):
-        if stream_name == "caches":
+        if stream_name == "events":
+            self._stream_id_gen.advance(token)
```

Comment:
`CacheInvalidationWorkerStore` seems like  a funny place for this. Shouldn't it be in `EventsWorkerStore`, if that's where `_stream_id_gen` lives?

(similarly for _backfill_id_gen)

Reply:
Woops, yes. Have moved those. I've just moved the advancing of the tokens, but not totally sure where the cache invalidation should live.